resume-buddy/src/main/java/com/google/sps/servlets/DeleteMatchDataServlet.java

```diff
@@ -0,0 +1,91 @@
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes Match data */
+@WebServlet("/delete-match-data")
+public class DeleteMatchDataServlet extends HttpServlet {
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    String revieweeEmail = userService.getCurrentUser().getEmail();
+
+    Query query = new Query("Match");
+    Filter revieweeFilter = new FilterPredicate("reviewee", FilterOperator.EQUAL, revieweeEmail);
+    query.setFilter(revieweeFilter);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    Entity entity = results.asSingleEntity();
+    String resumeBlobKey = (String) entity.getProperty("resumeBlobKey");
+    String matchID = (String) entity.getProperty("uuid");
+
+    // Deletes Match entity
+    datastore.delete(entity.getKey());
+
+    // Deletes matchID from user entities
+    query = new Query("User");
+    Filter matchIDFilter = new FilterPredicate("matchID", FilterOperator.EQUAL, matchID);
+    query.setFilter(matchIDFilter);
+    PreparedQuery usersResult = datastore.prepare(query);
+    for (Entity user : usersResult.asIterable()) {
+      user.setProperty("matchID", "");
+      datastore.put(user);
+    }
+
+    // Deletes all comments written on match's review page
+    query = new Query("Review-comments");
+    query.setFilter(matchIDFilter);
+    PreparedQuery commentsResult = datastore.prepare(query);
+    List<Key> commentKeys = new ArrayList<>();
+    for (Entity commentEntity : commentsResult.asIterable()) {
+      commentKeys.add(commentEntity.getKey());
+    }
+    datastore.delete(commentKeys);
+
+    // Deletes blob
+    int status = deleteBlob(resumeBlobKey);
+
+    response.setStatus(status);
+    response.sendRedirect("/index.html");
+  }
+
+  public int deleteBlob(String resumeBlobKey) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    boolean isBlobDeleted = false;
+
+    BlobKey blobKey = new BlobKey(resumeBlobKey);
+    blobstoreService.delete(blobKey);
+
+    BlobInfoFactory blobInfoFactory = new BlobInfoFactory();
+    isBlobDeleted = blobInfoFactory.loadBlobInfo(blobKey) == null;
+
+    if (isBlobDeleted) {
+      return 200; // 200: OK - The request is OK
+    } else {
+      return 400; // 400: Bad Request - The request cannot be fulfilled
+    }
+  }
```

Comment:
This is the same code in the DeleteBlobServlet. We want to avoid repetitive code. Is it possible to just call the DeleteBlobServlet to delete the blob for you, and then make sure the response it returns is recorded in the response returned by the `DeleteMatchDataServlet.doPut` function?

Reply:
I deleted DeleteBlobServlet and moved the codes in this function. We don't need DeleteBlob to be in a separate servlet because we always call it from other servlets.